cns/ipampoolmonitor/ipampoolmonitor.go

```diff
@@ -11,121 +13,193 @@ import (
 )
 
 var (
-	increasePoolSize = 1
-	decreasePoolSize = -1
-	doNothing        = 0
+	increasePoolSize = int64(1)
+	decreasePoolSize = int64(-1)
+	doNothing        = int64(0)
 )
 
 type CNSIPAMPoolMonitor struct {
-	initialized bool
+	initialized    bool
```

Comment:
I'm curious about the reason we need an initialized flag. Is there a way that the pool monitor can be hydrated on construction? That simplifies the rest of the methods, where we can assume that it's been initialized properly and that dependencies are there.